contracts/governance/Pauser.sol

```diff
@@ -0,0 +1,316 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.6.10;
+pragma experimental ABIEncoderV2;
+
+import {UpgradeableClaimable} from "../common/UpgradeableClaimable.sol";
+import {ITimelock} from "./interface/ITimelock.sol";
+import {IVoteToken} from "./interface/IVoteToken.sol";
+import {IOwnedUpgradeabilityProxy} from "../proxy/interface/IOwnedUpgradeabilityProxy.sol";
+import {ImplementationReference} from "../proxy/ImplementationReference.sol";
+import {IPauseableContract} from "../common/interface/IPauseableContract.sol";
+
+contract Pauser is UpgradeableClaimable {
+    // ================ WARNING ==================
+    // ===== THIS CONTRACT IS INITIALIZABLE ======
+    // === STORAGE VARIABLES ARE DECLARED BELOW ==
+    // REMOVAL OR REORDER OF VARIABLES WILL RESULT
+    // ========= IN STORAGE CORRUPTION ===========
+
+    // @notice The duration of voting on emergency pause
+    uint256 public votingPeriod;
+
+    // @notice The address of the TrustToken Protocol Timelock
+    ITimelock public timelock;
+
+    // @notice The address of the TrustToken Protocol Governor
+    IOwnedUpgradeabilityProxy public governor;
+
+    // @notice The address of the TrustToken governance token
+    IVoteToken public trustToken;
+
+    // @notice The address of the stkTRU voting token
+    IVoteToken public stkTRU;
+
+    // @notice The total number of requests
+    uint256 public requestCount;
+
+    // @notice The official record of all requests ever proposed
+    mapping(uint256 => PauseRequest) public requests;
+
+    // @notice The latest request for each requester
+    mapping(address => uint256) public latestRequestIds;
+
+    // ======= STORAGE DECLARATION END ============
+
+    // @notice The name of this contract
+    string public constant name = "TrueFi Pauser";
+
+    uint256 public constant EXECUTION_PERIOD = 1 days;
+
+    struct PauseRequest {
+        // @notice Unique id for looking up a request
+        uint256 id;
+        // @notice Creator of the request
+        address requester;
+        // @notice The timestamp that the request will be available for execution
+        uint256 eta;
+        // @notice the ordered list of target addresses of contracts to be paused
+        address[] targets;
+        // @notice The ordered list of functions to be called
+        // different types of proxies might require different types of pause functions
+        PausingMethod[] methods;
+        // @notice The block number at which voting begins: holders must delegate their votes prior to this block
+        uint256 startBlock;
+        // @notice The timestamp at which voting ends: votes must be cast prior to this timestamp
+        uint256 endTime;
```

Comment:
I notice that here we use an end *time* while in the very similar GovernorAlpha we use an end *block*. I'm not sure which is better, but we should probably make up our minds and pick one?